app/forms/vpc-router-route-common.tsx

```diff
@@ -76,16 +90,31 @@ export const RouteFormFields = ({ form, isDisabled }: RouteFormFieldsProps) => {
         items={toItems(destTypes)}
         placeholder="Select a destination type"
         required
+        onChange={() => {
+          form.setValue('destination.value', '')
+        }}
         disabled={isDisabled}
       />
-      <TextField
-        name="destination.value"
-        label="Destination value"
-        control={control}
-        placeholder="Enter a destination value"
-        required
-        disabled={isDisabled}
-      />
+      {destinationType === 'subnet' ? (
+        <ComboboxField
+          name="destination.value"
+          label="Destination value"
+          control={control}
+          items={vpcSubnets.map(({ name }) => ({ value: name, label: name }))}
+          placeholder="Select a destination value"
+          required
+          isDisabled={isDisabled}
+        />
+      ) : (
+        <TextField
+          name="destination.value"
+          label="Destination value"
+          control={control}
+          placeholder="Enter a destination value"
```

Comment:
I do like matching the API shape roughly, but maybe we can tweak some words to make this a little more human-feeling. Maybe it's enough to just change the placeholder based on the type selected, or maybe we want to go as far as changing the label? It would also be nice to get some kind of guidance about the right format for an IP network. I assume it's some kind of block, maybe we can give an example. 

<img width="479" alt="image" src="https://github.com/user-attachments/assets/7f657cc4-4208-4da3-897d-98caa1880d6f">